lib/queuePopulator/IngestionReader.js

```diff
@@ -140,22 +140,24 @@ class IngestionReader extends LogReader {
                 value: entry.value,
             }));
         } else {
+            let key;
+            let db;
             if (record.db === 'users..bucket') {
                 const keySplit = entry.key.split('..|..');
-                entry.key = `${keySplit[0]}..|..${this._targetZenkoBucket}`;
+                key = `${keySplit[0]}..|..${this._targetZenkoBucket}`;
             } else if (record.db === 'metastore') {
                 const keySplit = entry.key.split('/');
-                entry.key = `${keySplit[0]}/${this._targetZenkoBucket}`;
+                key = `${keySplit[0]}/${this._targetZenkoBucket}`;
             } else {
                 if (record.db === entry.key) {
-                    entry.key = this._targetZenkoBucket;
+                    key = this._targetZenkoBucket;
                 }
-                record.db = this._targetZenkoBucket;
+                db = this._targetZenkoBucket;
             }
             this._extensions.forEach(ext => ext.filter({
                 type: entry.type,
-                bucket: record.db,
-                key: entry.key,
+                bucket: db || record.db,
+                key: key || entry.key,
```

Comment:
I would rather check `key !== undefined` to avoid issues with potential defined but falsy keys.